config/initializers/rabl_init.rb

```diff
@@ -35,9 +35,37 @@ def data_name(data_token)
       # otherwise return super since v1 has object root (config.include_child_root = true)
       super
     end
+
+    def template_extensions_from_plugins
+      return [] unless @_options[:template]
+      Foreman::Plugin.all.flat_map {|plugin| plugin.api_template_extensions(@_options[:template]) }
+    end
+
+    def load_extensions_from_plugins
+      template_extensions_from_plugins.each do |extension|
+        extends extension unless @_settings[:extends].map { |e| e[:file] }.include?(extension)
+      end
+    end
+
+    def render(*args, &block)
+      load_extensions_from_plugins
+      super
+    end
+  end
+
+  module RablTemplateHandlerExt
+    def call(template)
+      source = template.source
+
+      %{ ::Rabl::Engine.new(#{source.inspect}, :template => '#{template.virtual_path}').
```

Comment:
Why do we need this?

Reply:
... to pass the template name (`template.virtual_path`) to the Engine so we can use it in `template_extensions_from_plugins `. An alternative to this would be `@_options[:template] || context_scope.virtual_path`.

Reply:
Any downside of the non-monkey patching version? That would be the option of my choice

Reply:
I agree that this monkey patching is not ideal, but I am not sure if `context_scope.virtual_path` is always the same as the template name. I think, it just refers to the controller + controller action. That is equal to the template name in most cases, but not always. Or am I missing something here?

Reply:
I see now why this is needed. Thanks for explanation